noxfile.py

```diff
@@ -261,23 +204,40 @@ def run_tests_with_dev_version_of(session: nox.Session, repository: str) -> None
             "https://pypi.anaconda.org/scientific-python-nightly-wheels/simple",
             "numpy",
         )
+    else:
+        session.install(f"git+{repository}")
+
     session.install(".[tests]")
     session.run(*pytest_command, *session.posargs)
 
 
-sphinx_commands: tuple[str, ...] = (
+doc_build_dir = "$READTHEDOCS_OUTPUT" if running_on_rtd else "docs/build/html"
+
+if running_on_rtd:
+    rtd_output_path = pathlib.Path(os.environ.get("READTHEDOCS_OUTPUT")) / "html"
+    rtd_output_path.mkdir(parents=True, exist_ok=True)
+    doc_build_dir = str(rtd_output_path)
+else:
+    doc_build_dir = "docs/build/html"
```

Comment:
Whenever we build the docs the build directory is `docs/_build` (with an underscore), not `docs/build`.  That's how the make files are constructed.

This build directory is a bit arbitrary though.  However, this would make the build directory different if you built with `make` versus `nox`.

Reply:
Also, this block can be streamlined more...

```suggestion
doc_build_dir = "docs/build/html"
if running_on_rtd:
    rtd_output_path = pathlib.Path(os.environ.get("READTHEDOCS_OUTPUT")) / "html"
    rtd_output_path.mkdir(parents=True, exist_ok=True)
    doc_build_dir = str(rtd_output_path)
```